tests/components/ipDetector.spec.ts

```diff
@@ -0,0 +1,72 @@
+import sinon from 'sinon';
+import { expect } from 'chai';
+import IpDetector from '../../src/components/ipDetector';
+import IIPDeviceManager from '../../src/interfaces/iIpDeviceManager';
+import DetectorOpts, { DetectionConvertion } from '../../src/interfaces/IDetectorOpts';
+import Logger from '../../src/utilitis/logger';
+import DeviceManagerMock from '../mocks/ipdevicemanager.mock';
+import * as msgpack from 'msgpack-lite';
+
+describe('IpDetector', () => {
+  let ipDetector: IpDetector;
+  let deviceManager: IIPDeviceManager;
+
+  beforeEach(() => {
+    deviceManager = new DeviceManagerMock();
+    ipDetector = new IpDetector(deviceManager);
+  });
+  describe('#init', () => {
+    let setIntervalSpy;
+    beforeEach(() => {
+      setIntervalSpy = sinon.spy(global, 'setInterval');
+    });
+    afterEach(() => {
+      setIntervalSpy.restore();
+      ipDetector.destroy();
+    });
+
+    describe('on detector not initialized', () => {
+      it('should set _detectorInitialized to true', async () => {
+        ipDetector = new IpDetector(deviceManager, {});
+        await ipDetector.init();
+        expect(ipDetector._detectorInitialized).to.equals(true);
+      });
+      it('should call setInterval', async () => {
+        ipDetector = new IpDetector(deviceManager, {});
+        await ipDetector.init();
+        expect(setIntervalSpy).to.have.been.calledOnce;
+      });
+    });
+    describe('on detector initialized', () => {
```

Comment:
Might be better to test that it actually what it asks for each interval, rather than it creates one.

Reply:
The test now checks that a detections event is fired at every interval.
Should I also add a test for checking that the ipDetector asks the grpcClient for detections every interval?

Reply:
No I think this is good enough.